Tests/NodeTests.cs

```diff
@@ -68,5 +69,84 @@ public void ValueBubblesDown ()
             p.Receive (Message.Event (b.Id, "change", "please work"));
             Assert.AreEqual ("please work", b.Value);
         }
+
+        class TestDiv : Div
+        {
+            public TestDiv (string id)
+            {
+                Id = id;
+            }
+        }
+
+        [TestMethod]
+        public void ReplaceChildrenWith ()
+        {
+            // As ReplaceChildrenWith calls ReplaceChildren
```

Comment:
I wrote a test suite that used randomization to get better coverage of the different use cases for replacement but wasn't sure if that would be considered overkill.